Publish godoc automatically on merge to master

pontus opened the pull request:
Hack to publish documentation to https://neicnordic.github.io/sda-pipeline/.

blankdots reviewed (commented):
Can we add the link to `README.md` ?

pontus commented:
I must have messed up pushes somewhere, fixed now (with the correct hack).

Not sure what readme to link to, added a link to the top of the repository for now (this is rendered at https://neicnordic.github.io/sda-pipeline/pkg/sda-pipeline/), is that good?
